2-copy-of-code/lesson-14/scripts/checkout.js

```diff
@@ -153,17 +153,27 @@ document.querySelectorAll('.js-save-link')
     link.addEventListener('click', () => {
       const productId = link.dataset.productId;
 
-      const container = document.querySelector(
-        `.js-cart-item-container-${productId}`
-      );
-      container.classList.remove('is-editing-quantity');
-
+      // Here's an example of a feature we can add: validation.
+      // Note: we need to move the quantity-related code up
+      // because if the new quantity is not valid, we should
+      // return early and NOT run the rest of the code. This
+      // technique is called an "early return".
       const quantityInput = document.querySelector(
         `.js-quantity-input-${productId}`
       );
       const newQuantity = Number(quantityInput.value);
+
+      if (newQuantity < 0 || newQuantity >= 1000) {
+        alert('Quantity must be at least 0 and less than 1000');
+        return;
+      }
       updateQuantity(productId, newQuantity);
 
+      const container = document.querySelector(
+        `.js-cart-item-container-${productId}`
+      );
+      container.classList.remove('is-editing-quantity');
+
```

Comment:
damn, been a while since i last touch the project so i dont remember all the codes. Looking at the above code, i would say it will remove the quantity once the number is out of bound (lets say it is 1001, which is more than the above limit - what we wouldnt want the user to do). Thinking of it as a 'quality of life'/UX code. They wont have to delete the number by themselves.